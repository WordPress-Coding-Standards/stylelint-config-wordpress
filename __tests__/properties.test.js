'use strict';

const fs = require( 'fs' ),
	config = require( '../' ),
	stylelint = require( 'stylelint' ),
	validCss = fs.readFileSync( './__tests__/properties-valid.css', 'utf-8' ),
	invalidCss = fs.readFileSync( './__tests__/properties-invalid.css', 'utf-8' );

describe( 'flags no warnings with valid properties css', () => {
	let result;

	beforeEach( () => {
		result = stylelint.lint({
			code: validCss,
			config,
		});
	});

	it( 'did not error', () => {
		return result.then( data => (
			expect( data.errored ).toBeFalsy()
		) );
	});

	it( 'flags no warnings', () => {
		return result.then( data => (
			expect( data.results[0].warnings ).toHaveLength( 0 )
		) );
	});
});

describe( 'flags warnings with invalid properties css', () => {
	let result;

	beforeEach( () => {
		result = stylelint.lint({
			code: invalidCss,
			config,
		});
	});

	it( 'did error', () => {
		return result.then( data => (
			expect( data.errored ).toBeTruthy()
		) );
	});

	it( 'flags correct number of warnings', () => {
		return result.then( data => (
			expect( data.results[0].warnings ).toHaveLength( 7 )
		) );
	});

	it( 'snapshot matches warnings', () => {
		return result.then( data => (
			expect( data.results[0].warnings ).toMatchSnapshot()
		) );
	});
});
