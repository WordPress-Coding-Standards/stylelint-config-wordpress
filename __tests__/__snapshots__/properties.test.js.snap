// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flags warnings with invalid properties css snapshot matches warnings 1`] = `
Array [
  Object {
    "column": 13,
    "line": 2,
    "rule": "color-hex-case",
    "severity": "error",
    "text": "Expected \\"#FFFFFF\\" to be \\"#ffffff\\" (color-hex-case)",
  },
  Object {
    "column": 13,
    "line": 2,
    "rule": "color-hex-length",
    "severity": "error",
    "text": "Expected \\"#FFFFFF\\" to be \\"#FFF\\" (color-hex-length)",
  },
  Object {
    "column": 2,
    "line": 5,
    "rule": "declaration-block-no-shorthand-property-overrides",
    "severity": "error",
    "text": "Unexpected shorthand \\"margin\\" after \\"margin-left\\" (declaration-block-no-shorthand-property-overrides)",
  },
  Object {
    "column": 13,
    "line": 2,
    "rule": "declaration-colon-space-after",
    "severity": "error",
    "text": "Expected single space after \\":\\" with a single-line declaration (declaration-colon-space-after)",
  },
  Object {
    "column": 2,
    "line": 6,
    "rule": "property-no-unknown",
    "severity": "error",
    "text": "Unexpected unknown property \\"argin\\" (property-no-unknown)",
  },
  Object {
    "column": 15,
    "line": 4,
    "rule": "unit-case",
    "severity": "error",
    "text": "Expected \\"PX\\" to be \\"px\\" (unit-case)",
  },
  Object {
    "column": 11,
    "line": 3,
    "rule": "value-keyword-case",
    "severity": "error",
    "text": "Expected \\"BLOCK\\" to be \\"block\\" (value-keyword-case)",
  },
]
`;
